app/src/androidTest/java/com/example/rwothoromo/developers/GithubUserProfileTest.java

```diff
@@ -0,0 +1,125 @@
+package com.example.rwothoromo.developers;
+
+import android.content.Context;
+import android.content.Intent;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.espresso.Espresso;
+import android.support.test.espresso.idling.CountingIdlingResource;
+import android.support.test.espresso.intent.Intents;
+import android.support.test.espresso.matcher.ViewMatchers;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.runner.AndroidJUnit4;
+
+import com.example.rwothoromo.developers.view.GithubUserProfile;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.intent.matcher.IntentMatchers.hasAction;
+import static android.support.test.espresso.intent.matcher.IntentMatchers.hasExtra;
+import static android.support.test.espresso.matcher.ViewMatchers.withContentDescription;
+import static android.support.test.espresso.matcher.ViewMatchers.withEffectiveVisibility;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static com.example.rwothoromo.developers.constants.Constants.EXTRA_SHARE_INTENT_TEXT;
+import static org.hamcrest.CoreMatchers.allOf;
+import static org.hamcrest.CoreMatchers.is;
+
+/**
+ * GithubUserProfile test.
+ */
+@RunWith(AndroidJUnit4.class)
+public class GithubUserProfileTest {
+
+    /**
+     * GithubUserProfile test rule.
+     */
+    @Rule
+    public ActivityTestRule<GithubUserProfile> mGithubUserProfileTestRule
+            = new ActivityTestRule<GithubUserProfile>(GithubUserProfile.class) {
+
+        @Override
+        protected Intent getActivityIntent() {
+            Context targetContext = InstrumentationRegistry.getTargetContext();
+            Intent intent = new Intent(targetContext, GithubUserProfile.class);
+            intent.setAction(Intent.ACTION_SEND);
+            intent.putExtra(Intent.EXTRA_TEXT, EXTRA_SHARE_INTENT_TEXT);
+            return intent;
+        }
+    };
+
+    private CountingIdlingResource countingResource;
+
+    /**
+     * Register any resource that needs to be synchronized with Espresso before the test is run.
+     *
+     * @throws Exception when method fails
+     */
+    @Before
+    public void setUp() throws Exception {
```

Comment:
I am not sure you need to register an idling resource for this too, it does not involve any network calls.

Reply:
Thank you @idrisadetunmbi, there is actually no need for an idling resource here, so I'll remove it.